### ISSUE TITLE
Force ALT text on images

### ISSUE THREAD

**Issue by curtismchale** [proudcity/wp-proudcity#2546]
### Deploy ALT Text plugin

- [staging](https://console.cloud.google.com/cloud-build/builds;region=global/9b59ee3a-f9ad-4fb9-b7c5-d7893b7542bc?project=proudcity-1184)
- [master](https://console.cloud.google.com/cloud-build/builds;region=global/c3083d37-36be-4c76-aece-5d00fe7151a0?project=proudcity-1184)

## Source

CM

## Describe the bug
ALT text should be forced on all instances of adding an image.

### Adding image to Media Gallery

URL: https://beta.proudcity.com/wp-admin/upload.php

When you upload an image it should force you to put in ALT text in.

### Inserting image to WP via the Classic Editor

When you choose an image from the media gallery it should prompt you for ALT text if there is none.

![CleanShot 2024-05-02 at 13 16 33](https://github.com/proudcity/wp-proudcity/assets/74561/ce9bc549-f362-4a8a-908b-93fccca2d38d)

![CleanShot 2024-05-02 at 13 17 28](https://github.com/proudcity/wp-proudcity/assets/74561/6e3db00f-8e6f-4c01-b0c7-196e863ffb64)

### With Site Origin widgets

When you choose media it should auto-populate the existing ALT text and if there is none it should **not** allow you to insert the image.

**Image Widget**
![CleanShot 2024-05-02 at 13 18 43](https://github.com/proudcity/wp-proudcity/assets/74561/c9ca015d-a456-45f9-b0a4-9897668ecc7d)

**Image Set Widget**

All images in the set should have ALT text and any that don't should have the issue highlighted if the user attempts to insert the content.

![CleanShot 2024-05-02 at 13 20 44](https://github.com/proudcity/wp-proudcity/assets/74561/eb7e9562-2957-4f5d-902b-bbdea1f7ba19)

## QA

Notes:

* 

Links:

*

**Comment by kevindherman** [comment]
@curtismchale additional widgets that use images and should have alt text requirements:

- Hero unit 
- Gallery

**Comment by kevindherman** [comment]
All of these files come from the media library so my hope is that if we start with forcing alt text to be required when uploading the original, then alt text is addressed in all applications of the file.

**Comment by curtismchale** [comment]
@kevindherman we can force ALT text in the media gallery when someone selects an image with the [Fix ALT Text](https://wordpress.org/plugins/fix-alt-text/) plugin.

This doesn't handle the ALT text in widgets as it won't fill in the text when you add an image but it will stop anyone from picking an image without ALT text so it would force older images to get the text and any new image would get it as well.

I'll have to build something custom to work with SiteOrigin widgets. Should I do that or stop here for now?

**Comment by kevindherman** [comment]
@curtismchale can this at least be deployed to Petaluma? 

I think getting this part on Petaluma would be a nice win then circle back for the rest with breathing room.

**Comment by curtismchale** [comment]
@kevindherman is there a reason we wouldn't deploy it to all sites?

**Comment by kevindherman** [comment]
@curtismchale all sounds better.

**Comment by kevindherman** [comment]
@curtismchale @lukefretwell the plugin looks great and is definitely an improvement from what we had before. 

I really like the reporting/fixing feature to help solve the problem.

**Comment by curtismchale** [comment]
@kevindherman it's on Beta now and getting ready to go to all sites.

**Comment by kevindherman** [comment]
@curtismchale this is great and I actually think this does everything we need. 

1. I added media in visual editor grabbing image file that was already uploaded and had no alt text, it required me to add it
2. I added an image set and did the same as above
3. I uploaded a new image for the image set and needed to add the alt text
4. I added an image widget with the same experience.

I'm not sure the widget level alt text is needed but we can revisit.

It may be part of a followup but the plugin's reporting/fixing is really nice and could go under tools in our admin. I didn't see where to access and play around with those features so lets not have that hold this up. 

Huge improvement as is. @lukefretwell

**Comment by curtismchale** [comment]
Deployed the plugin

Composer: https://github.com/proudcity/wp-proudcity/releases/tag/2024.05.07.1214
Build: https://github.com/proudcity/proud-recipes/releases/tag/2024.05.07.1214
